libbindgen/src/ir/context.rs

```diff
@@ -441,6 +443,40 @@ impl<'ctx> BindgenContext<'ctx> {
         ret
     }
 
+    /// This function trying to find any dangling references inside of `items`
+    fn assert_no_dangling_references(&self) {
+        if cfg!(debug_assertions) {
```

Comment:
With the changes requested below, the contents of this if branch can now simply be:

```rust
if cfg!(debug_assertions) {
    for _ in self.assert_no_dangling_item_traversal() {
        // The iterator's next method does the asserting for us.
    }
}
```